devito/tools/dtypes_lowering.py

```diff
@@ -133,6 +133,19 @@ def dtype_to_cstr(dtype):
 
 def dtype_to_ctype(dtype):
     """Translate numpy.dtype into a ctypes type."""
+    if isinstance(dtype, CustomDtype):
+        return dtype
+
+    # Complex data
+    if np.issubdtype(dtype, np.complexfloating):
+        rtype = dtype(0).real.__class__
+        from devito import configuration
+        make = configuration['compiler']._complex_ctype
```

Comment:
you really can't use global information here

Reply:
It's not global because this is called within the switchconfig

Reply:
from that code path above yes, but from another one?

we shouldn't access `configuration` in these remote places